contexts/MembershipContext/src/Domain/Model/Application.php

```diff
@@ -166,4 +166,16 @@ public function isDeleted(): bool {
 		return $this->isDeleted;
 	}
 
+	public function setFirstPaymentDate( string $firstPaymentDate ) {
+		/** @var PayPalPayment $payPalPayment */
+		$payPalPayment = $this->getPayment()->getPaymentMethod();
+		$payPalData = $payPalPayment->getPayPalData();
+		if ( $payPalData === null ) {
+			$payPalData = new PayPalData();
+		}
+
+		$payPalData->setFirstPaymentDate( $firstPaymentDate );
+		$this->addPayPalData( $payPalData );
```

Comment:
I tried out making the field non-nullable and it worked: https://github.com/wmde/FundraisingFrontend/pull/905

The other question still stands. Furthermore the expectorations around PayPalData are not clear to me. Looks like if you try to get one of its values when it is not initialized yet, you will get an error. And finally the whole PaymentMethod thing comes over as weirdly structured. Not sure how it could be better. Perhaps @wiese with his fresh pair of eyes has an idea?

Not liking the addPayPalData in Donation and Application. Think this should be in the payment method itself. Looking at where this method is used, that should be easy enough to do. (In another PR, since that really is not part of the task here.)